### PR TITLE
feat(models): add InternVL-U VLM support

### PR DESCRIPTION
**Pull request by vansin**
## Summary

Add full [InternVL-U](https://huggingface.co/OpenGVLab/InternVL-U) support to SGLang — both text understanding (srt) and image generation (multimodal_gen).

InternVL-U is a 4B unified multimodal model by [OpenGVLab](https://github.com/OpenGVLab/InternVL-U) combining InternViT-300M + Qwen3-1.7B VLM + MMDiT 1.7B diffusion decoder ([paper](https://arxiv.org/abs/2603.09877)).

### Changes (4 files)

| File | Description |
|------|-------------|
| `python/sglang/srt/models/internvl_u.py` | VLM model — extends InternVLChatModel, filters diffusion weights |
| `python/sglang/srt/configs/model_config.py` | Register InternVLUChatModel as multimodal architecture |
| `python/sglang/multimodal_gen/runtime/pipelines/internvl_u.py` | Unified VLM+DiT+VAE pipeline with torch.compile |
| `docs/.../multimodal_language_models.md` | Add InternVL-U to supported models |

### Design

**VLM (srt)**: Inherits InternVLChatModel (same InternViT + Qwen3 architecture). Only overrides `load_weights` to skip `special_token_embedding`.

**Image gen (multimodal_gen)**: Loads full model as single unit. `InternVLUExecutionStage` handles text → VLM hidden states → DiT denoising → VAE decode → image. OpenAI `/v1/images/generations` compatible.

### Text Understanding

| Test | SGLang | PyTorch | Speedup |
|------|--------|---------|---------|
| Short (~30 tok) | 0.56s | 1.27s | **2.25x** |
| Chinese (~100 tok) | 1.00s | 3.82s | **3.84x** |
| Code (~200 tok) | 1.37s | 5.03s | **3.66x** |
| Long (~300 tok) | 2.08s | 2.97s | **1.42x** |

Avg **2.8x** speedup, 145 tok/s peak, 4/5 outputs identical at T=0

### Image Generation

| Resolution | PyTorch | SGLang | Speedup |
|-----------|---------|--------|---------|
| 256x256 | 1.98s | 1.59s | **1.25x** |
| 512x512 | 3.32s | 2.54s | **1.31x** |

Pixel-identical precision (PSNR=infinity)

## Test Plan

- [x] VLM: 5 unit tests passing (math, code, knowledge, Chinese, determinism)
- [x] Image gen: /v1/images/generations API verified
- [x] bench_one_batch prefill logits verified
- [x] Precision alignment verified
- [x] Speed benchmarks on A100-80GB

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/pipelines/internvl_u.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+from sglang.multimodal_gen.runtime.pipelines_core.stages import PipelineStage
+from sglang.multimodal_gen.runtime.server_args import ServerArgs
+from sglang.multimodal_gen.runtime.utils.logging_utils import init_logger
+
+logger = init_logger(__name__)
+
+
+def _setup_internvlu_imports():
+    """Ensure InternVL-U source code is importable."""
+    src_path = os.environ.get("INTERNVLU_SRC_PATH")
```
Comment (ping1jing2) [inline python/sglang/multimodal_gen/runtime/pipelines/internvl_u.py]: please use `envs.get` rather than `os.environ.get` in sglang, refer to `python\sglang\multimodal_gen\envs.py`

File: python/sglang/multimodal_gen/runtime/pipelines/internvl_u.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        sys.path.insert(0, src_path)
+
+    # Apply flash_attn SDPA patch if needed
+    patch_path = os.environ.get("FLASH_ATTN_PATCH_PATH")
+    if patch_path and patch_path not in sys.path:
+        sys.path.insert(0, patch_path)
+    try:
+        import flash_attn_patch  # noqa: F401
+    except ImportError:
+        pass
```
Comment (ping1jing2) [inline python/sglang/multimodal_gen/runtime/pipelines/internvl_u.py]: it might be better to riase a warning msg here if we want users use flash_attn_patch

File: python/sglang/multimodal_gen/runtime/pipelines/internvl_u.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+class InternVLUExecutionStage(PipelineStage):
+    """Execute full InternVL-U pipeline: prompt -> VLM -> DiT -> image."""
+
+    def __init__(self, pipe: Any, processor: Any):
+        super().__init__()
+        self.pipe = pipe
+        self.processor = processor
+
+    def forward(self, batch: Req, server_args: ServerArgs) -> Req:
+        prompt = getattr(batch, "prompt", "") or ""
```
Comment (ping1jing2) [inline python/sglang/multimodal_gen/runtime/pipelines/internvl_u.py]: please don't use `getattr/setattr` in codebase, it's hard to trace the code

File: python/sglang/srt/models/internvl_u.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+"""
+
+from typing import Iterable, Tuple
+
+import torch
+
+from sglang.srt.models.internvl import InternVLChatModel
+
+
+class InternVLUChatModel(InternVLChatModel):
```
Comment (ping1jing2) [inline python/sglang/srt/models/internvl_u.py:39]: could you please add a UT for this model

File: python/sglang/multimodal_gen/runtime/pipelines/internvl_u.py [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+    ) -> Dict[str, Any]:
+        _setup_internvlu_imports()
+
+        model_path = self.model_path
+        dtype = torch.float16
+
+        logger.info("Loading InternVL-U pipeline from %s", model_path)
+
+        from internvlu import InternVLUPipeline
+        from internvlu.processing_internvlu import InternVLUProcessor
```
Comment (ping1jing2) [inline python/sglang/multimodal_gen/runtime/pipelines/internvl_u.py]: shall we move these imports to the top of this file
